src/riscv/lib/src/state_context/projection.rs

```diff
@@ -108,7 +108,11 @@ pub trait Projection {
     /// offset to an address of the subject value that would give you the address of the target
     /// value. This is exclusive to the [`crate::state_backend::owned_backend::Owned`] state
     /// backend.
-    fn owned_pointer_offset<MC: MemoryConfig>(param: Self::Parameter) -> i32;
+    ///
+    /// When using this value, care should be taken when converting to signed offsets.
+    /// In particular, the offset represents a _positive_ value and this must be upheld
+    /// when converting to signed values.
+    fn owned_pointer_offset<MC: MemoryConfig>(param: Self::Parameter) -> usize;
```

Comment:
Could you introduce a dedicated type to encode the right semantics? 
Or perhaps you could re-use [`Offset32`](https://docs.rs/cranelift-codegen/latest/cranelift_codegen/ir/immediates/struct.Offset32.html).

Reply:
absolutely, wasn't sure if it would be overly complex or not. But makes sense to indeed

Reply:
have introduced `ProjectionOffset` now, which can add itself/usize.